### PR TITLE
adding canary docs

### PR DESCRIPTION
**Pull request by krishnacpuvvada**
# What does this PR do ?

Add a one line overview of what this PR aims to accomplish.

**Collection**: [ASR]

# Changelog 
- Adding docs for canary

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
 - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [x] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: docs/source/asr/configs.rst [hunk 1]
```diff
@@ -1120,3 +1120,160 @@ Depending on the type of model, there may be extra steps that must be performed
 
 * CTC Models - `Examples directory for CTC Models <https://github.com/NVIDIA/NeMo/blob/stable/examples/asr/asr_ctc/README.md>`_
 * RNN Transducer Models - `Examples directory for Transducer Models <https://github.com/NVIDIA/NeMo/blob/stable/examples/asr/asr_transducer/README.md>`_
+
+Multi-task AED Models
```
Comment (nithinraok) [inline docs/source/asr/configs.rst:1124]: Is the following information present in the docs:
* What is Canary model 
* How to load a canary model and transcribe? 
* How to finetune a Canary model to adapt to new languages
* Same languages but new data -> tokenizer update
* Recommended hyper parameters for finetuning

**Review by nithinraok — CHANGES_REQUESTED** [review]

File: docs/source/asr/configs.rst [hunk 2]
```diff
@@ -1120,3 +1120,160 @@ Depending on the type of model, there may be extra steps that must be performed
...
+.. code-block:: yaml
+
+    train_ds:
+      use_lhotse: true
+      input_cfg: "<path to lhotse config>"
+      batch_duration: 600
+      max_duration: 40
+      use_bucketing: True
+      num_buckets: 30
+      bucket_duration_bins: [3.79, 4.82, 5.688, ..., 35.827]
```
Comment (titu1994) [inline docs/source/asr/configs.rst:1181]: Provide info / link to calcualte bucket duration bins (ir just update below sentence to mention that to get the list of floats, follow this link)

File: docs/source/asr/configs.rst [hunk 3]
```diff
@@ -1120,3 +1120,160 @@ Depending on the type of model, there may be extra steps that must be performed
...
+      text_field: "answer"
+      lang_field: "target_lang"
+
+For more details about Lhotse-based dataset specification, refer to `Lhotse Dataloading <./datasets.html#_hotse_dataloading>`__
+
+Special Tokens Configuration
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The model uses special tokens for task control and language specification. These can be configured during tokenizer creation or loaded from an existing tokenizer:
```
Comment (titu1994) [inline docs/source/asr/configs.rst:1190]: we need docs how to construct this special tokenizer (either here or somewhere below and link to that section)

File: docs/source/asr/configs.rst [hunk 4]
```diff
@@ -1120,3 +1120,160 @@ Depending on the type of model, there may be extra steps that must be performed
...
+      asr_enc_hidden: 1024    # Hidden size for ASR encoder
+      lm_dec_hidden: 1024     # Hidden size for transformer decoder
+      text_field: "answer"    # Field name for ground truth in manfiest
+      lang_field: "target_lang"    # Field name for output language in manifest
+
+These parameters are referenced throughout the config using OmegaConf interpolation (${...}) to maintain consistency across different components of the model.
+
+
+Transformer Decoder
+~~~~~~~~~~~~~~~~~
```
Comment (titu1994) [inline docs/source/asr/configs.rst:1249]:
```suggestion
~~~~~~~~~~~~~~~~~~~
```

File: docs/source/asr/images/aed_model.png
Comment (titu1994) [inline docs/source/asr/images/aed_model.png:1]: No picture upload. Upload to latest nemo release and update url

**Review by titu1994 — COMMENTED** [review]
Overall looks like a good start but needs more details for setting up internals

**Comment by titu1994** [comment]
@krishnacpuvvada  please address and merge

File: docs/source/asr/configs.rst (hunk 2 above)
Comment (krishnacpuvvada) [reply docs/source/asr/configs.rst:1181]: done.

File: docs/source/asr/configs.rst (hunk 4 above)
Comment (krishnacpuvvada) [reply docs/source/asr/configs.rst:1249]: done

**Pull request closed without merging**